databases/week3/createTables.sql

```diff
@@ -0,0 +1,147 @@
+SELECT * FROM reservation;
+
+-- Add a new reservation
+INSERT INTO reservation (number_of_guests, meal_id, created_date) VALUES (8, 9, "2020-08-17");
+
+-- Get a reservation with any id, fx 1
+SELECT * FROM reservation WHERE id = 1;
+
+-- Update a reservation with any id, fx 1. Update any attribute fx the title or multiple attributes
+UPDATE reservation SET number_of_guests = 11, meal_id = 8 WHERE id = 1;
+
+-- Delete a reservation with any id, fx 1
+DELETE FROM reservation WHERE id = 6;
+
+
+-- Review
+INSERT INTO review (title, description, meal_id, Stars, created_date) 
+VALUES ('best dinner', 'I like it very much', 4, 5, "2019-9-28");
+INSERT INTO review (title, description, meal_id, Stars, created_date) 
+VALUES ('Average dinner', 'I like it very much', 3, 5, "2019-11-8");
+INSERT INTO review (title, description, meal_id, Stars, created_date) 
+VALUES ('Excellent dinner', 'I like it very much', 2, 5, "2019-1-28");
+INSERT INTO review (title, description, meal_id, Stars, created_date) 
+VALUES ('Worst dinner', 'I do not like it', 1, 1, "2019-10-2");
+
+
+-- Get all reviews
+SELECT * FROM review;
+
+-- Add a new review
+INSERT INTO review (title, description, meal_id, Stars, created_date) 
+VALUES ('nice dinner', 'I like it', 1, 4, "2019-1-2");
+
+-- Get a review with any id, fx 1
+SELECT * FROM review WHERE id = 1;
+
+-- Update a review with any id, fx 1. Update any attribute fx the title or multiple attributes
+UPDATE review SET title = 'nice', stars = 3 WHERE id = 1;
+
+-- Delete a review with any id, fx 1
+DELETE FROM review WHERE id = 5;
+
+
+-- Additional queries
+
+-- Get meals that has a price smaller than a specific price fx 90
+SELECT * FROM Meal WHERE Price < 4;
+
+-- Get meals that still has available reservations
+SELECT meal.title, meal.max_reservations, reservation.number_of_guests FROM meal JOIN reservation ON meal.id = reservation.meal_id
+WHERE meal.max_reservations >  reservation.number_of_guests;
+
+-- Get meals that partially match a title. Rød grød med will match the meal with the title Rød grød med fløde
+SELECT * FROM meal WHERE title LIKE '%Rød grød med fløde%'; 
+
+-- Get meals that has been created between two dates
+SELECT * FROM meal
+WHERE created_date BETWEEN '2019-01-5' AND '2030-10-20';
+
+-- Get only specific number of meals fx return only 5 meals
+SELECT * FROM Meal LIMIT 5;
+
+-- Get the meals that have good reviews
+SELECT meal.title, meal.location, review.stars AS star_rating FROM meal 
+JOIN review ON meal.id = review.meal_id WHERE stars > 3;
+
+-- Get reservations for a specific meal sorted by created_date
+SELECT meal.title, reservation.created_date, reservation.number_of_guests
+FROM meal JOIN reservation ON meal.id = reservation.meal_id
+WHERE meal.title LIKE '%soup%'
+ORDER BY reservation.created_date;
+
+-- Sort all meals by average number of stars in the reviews
+SELECT meal.title, AVG(review.stars) AS stars FROM meal JOIN review ON meal.id = review.meal_id 
+GROUP BY meal.title
```

Comment:
Great!. Although be careful about using meal.title for group by. You have two different meals with a title of "pizza" which will mean that they will be group incorrectly. It's better to use meal.id for group by

Reply:
changed meal.id for GROUP BY